src/GENIEManager.cpp

```diff
@@ -0,0 +1,193 @@
+           << G4endl
+           << "Current list of supported genie formats:" << G4endl
+           << " - rootracker" << G4endl;
+    abort();
+  } 
+ 
+  if (!inputFile_.empty())
+  {
+    isInitialized=0;
+    G4cout << "Configuring to Read From ROOT File..." << G4endl;
+    // Reading the root file to extract the info
+    if (tree_ == 0) {
+      f_ = (TFile*)gROOT->GetListOfFiles()->FindObject(inputFile_);
+      if (!f_ || !f_->IsOpen()) {
+        f_ = new TFile(inputFile_);
+      }
+      if(f_->IsZombie()){
+        return isInitialized ;
+      }
+      f_->GetObject(treeName_,tree_);
+      if (!(tree_ == 0)) {
+        this->SetBranches();
+      } else {
+        G4cerr << "tree_ is a null pointer" << G4endl;
+        abort;
+      }
+    }
+    isInitialized=1;
+    return isInitialized;
+  }
+  else {
+    G4cout << "Generating from ROOT is not selected!  Continuing..."<< G4endl;
+    isInitialized=0;
+    return isInitialized;
+  }
+
+}
+
+
+void GENIEManager::SetBranches() {
+  if (!(tree_ == 0)){
+    // Creating Root Branches
+    if (treeName_=="gRooTracker") {
+      tree_->SetBranchAddress("EvtNum",           &event_,            &b_event);
+      //tree_->SetBranchAddress("EvtFlags",        flags_,            &b_flags);
+      //tree_->SetBranchAddress("EvtCode",         code_,             &b_code);
+      //tree_->SetBranchAddress("EvtXSec",         xsec_,             &b_xsec);
+      //tree_->SetBranchAddress("EvtDXSec",        dxsec_,            &b_dxsec);
+      tree_->SetBranchAddress("EvtWght",          &weight_,           &b_weight);
+      tree_->SetBranchAddress("EvtProb",          &prob_,             &b_prob);
+      tree_->SetBranchAddress("EvtVtx",           &vtx_,              &b_vtx);
+      tree_->SetBranchAddress("StdHepN",          &NParticles_,       &b_NParticles);
+      tree_->SetBranchAddress("StdHepPdg",        &pdg_,              &b_pdg);
+      tree_->SetBranchAddress("StdHepStatus",     &Status,            &brStatus   );
+      //tree_->SetBranchAddress("StdHepRescat",    rescat_,           &b_rescat);
+      //tree_->SetBranchAddress("StdHepX4",        X4_,               &b_X4);
+      tree_->SetBranchAddress("StdHepP4",         &P4_,               &b_P4);
+      //tree_->SetBranchAddress("StdHepPolz",      polz_,             &b_polz);
+      tree_->SetBranchAddress("StdHepFm",         &FirstMother,	    &brFirstMother   );
+      tree_->SetBranchAddress("StdHepLm",         &LastMother,	    &brLastMother   );
+      tree_->SetBranchAddress("StdHepFd",         &FirstDaughter,     &brFirstDaughter   );
+      tree_->SetBranchAddress("StdHepLd",         &LastDaughter,      &brLastDaughter   );
+    }
+  }
+  else{
+    G4cerr << "tree_ is a null pointer" << G4endl;
+    abort;
+  }
+
+
+}
+
+Int_t GENIEManager::GetNEntries() {
+  return tree_->GetEntriesFast();
+}
+
+
+void GENIEManager::Cd(){
+  if(f_)
+    f_->cd();
```

Comment:
Two comments:  Why not use a one-line statement here?  Insert a space between "if" and "(".